### ISSUE TITLE
`private_over_fileprivate` produces a false positive when using actor

### ISSUE THREAD

**Issue by jbttn** [realm/SwiftLint#5489]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`private_over_fileprivate` produces a false positive when using `actor`.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? `0.53.0`
* Installation method used (Homebrew, CocoaPods, building from source, etc)? `Homebrew`
* Paste your configuration file:

```yml
disabled_rules:
  - todo

file_length:
  ignore_comment_only_lines: true
function_body_length: 75
identifier_name:
  excluded:
    - i
    - j
    - id
    - db
line_length:
  ignores_comments: true
type_name:
  excluded:
    - ID
    - L
```

> Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?

No

> Which Xcode version are you using (check `xcodebuild -version`)?

Xcode 15.3
Build version 15E204a

> Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

  ```swift
  import SwiftUI
  
  extension EnvironmentValues {
      var foo: Foo { .shared }
  }
  
  actor Foo {
    fileprivate static let shared = Foo() // Triggers the warning
  }
  
  //class Foo {
  //  fileprivate static let shared = Foo() // Does not trigger the warning
  //}
  ```
